session_rpcserver.go

```diff
@@ -214,15 +215,92 @@ func (s *sessionRpcServer) resumeSession(sess *session.Session) error {
 		return nil
 	}
 
-	authData := []byte(fmt.Sprintf("%s: %s", HeaderMacaroon, mac))
+	var (
+		isFirstConn             = sess.RemotePublicKey == nil
+		onNewStatus             func(s mailbox.ServerConnStatus)
+		firstConnectionDeadline *time.Timer
+		firstConnTimout         = make(chan struct{})
+		exitFirstConnDeadline   = make(chan struct{})
+	)
+
+	// If this is the first time the session is being spun up then we will
+	// kick off a timer to revoke the session after a timeout unless an
+	// initial connection is made. We identify such a session as one that
+	// we do not yet have a static remote pub key for.
+	if isFirstConn {
+		deadlineDuration := sess.CreatedAt.Add(
+			s.cfg.firstConnectionDeadline,
+		)
+
+		if deadlineDuration.Before(time.Now()) {
+			log.Debugf("Deadline for session %s has already "+
+				"passed. Revoking session", pubKeyBytes)
+
+			err = s.db.RevokeSession(pubKey)
+			if err != nil {
+				log.Debugf("error revoking session: "+"%v", err)
+			}
+		}
+
+		// Start the deadline timer.
+		firstConnectionDeadline = time.NewTimer(
```

Comment:
I think we could use `time.AfterFunc()` here that closes the `firstConnTimeout`. Then in the `onNewStatus` callback we only stop that timer and we don't need a whole other goroutine (the debug comment could be moved into the `onNewStatus` callback).

Reply:
omg yeeeesss!!! thank you :) updated!